tests/codex/testvalidation.nim

```diff
@@ -40,13 +60,45 @@ asyncchecksuite "validation":
 
   test "the list of slots that it's monitoring is empty initially":
     check validation.slots.len == 0
+  
+  test "maxSlots in ValidationParams must be greater than 0":
+    let maxSlots = 0
+    let validationParams: ?!ValidationParams = ValidationParams.init(maxSlots = maxSlots, partitionSize = 1, partitionIndex = 0)
+    check validationParams.isFailure == true
+    check validationParams.error.msg == fmt"maxSlots must be greater than 0! (got: {maxSlots = })"
+
+  for partitionSize in [-100, -1, 0]:
+    test fmt"initializing ValidationParams fails when partitionSize has value our of range (testing for {partitionSize = })":
+      let validationParams: ?!ValidationParams = ValidationParams.init(maxSlots, partitionSize = partitionSize, partitionIndex = 0)
+      check validationParams.isFailure == true
+      check validationParams.error.msg == fmt"Partition size must be greater than 0! (got: {partitionSize = })"
+  
+  test fmt"initializing ValidationParams fails when partitionIndex is negative":
+    let partitionIndex = -1
+    let validationParams: ?!ValidationParams = ValidationParams.init(maxSlots, partitionSize = 1, partitionIndex = partitionIndex)
+    check validationParams.isFailure == true
+    check validationParams.error.msg == fmt"Partition index must be greater than or equal to 0! (got: {partitionIndex = })"
+
+  for (partitionSize, partitionIndex) in [(100, 100), (100, 101)]:
+    test fmt"initializing ValidationParams fails when partitionIndex is greater than or equal to partitionSize  (testing for {partitionIndex = }, {partitionSize = })":
+      let validationParams: ?!ValidationParams = ValidationParams.init(maxSlots, partitionSize = partitionSize, partitionIndex = partitionIndex)
+      check validationParams.isFailure == true
+      check validationParams.error.msg == fmt"The value of the partition index must be less than partition size! (got: {partitionIndex = }, {partitionSize = })"
 
   test "when a slot is filled on chain, it is added to the list":
     await market.fillSlot(slot.request.id, slot.slotIndex, proof, collateral)
     check validation.slots == @[slot.id]
+  
+  test "slot is not observed if it is not in the partition":
+    let validationParams = initValidationParams(maxSlots, partitionSize, partitionIndex + 1)
+    let validation = createValidation(clock, market, validationParams)
+    await validation.start()
+    await market.fillSlot(slot.request.id, slot.slotIndex, proof, collateral)
+    await validation.stop()
```

Comment:
I am creating a new instance of the validation here because it needs customised validation params. Thus I am overwriting the one already created in setup, and so the `stop` in `teardown` will not work on this one.

Reply:
Ah good point, didn't notice that.